src/couch_mrview_util.erl

```diff
@@ -341,18 +323,9 @@ all_docs_reduce_to_count(Reductions) ->
 reduce_to_count(nil) ->
     0;
 reduce_to_count(Reductions) ->
-    Reduce = fun
-        (reduce, KVs) ->
-            Counts = [
-                case V of {dups, Vals} -> length(Vals); _ -> 1 end
-                || {_,V} <- KVs
-            ],
-            {lists:sum(Counts), []};
-        (rereduce, Reds) ->
-            {lists:sum([Count0 || {Count0, _} <- Reds]), []}
-    end,
-    {Count, _} = couch_btree:final_reduce(Reduce, Reductions),
-    Count.
+    CountsReduceFun = fun counts_reduce/2,
+    FinalReduction = couch_btree:final_reduce(CountsReduceFun, Reductions),
+    get_counts(FinalReduction).
```

Comment:
A bit naming issue: `reduce_to_count` means that we expect singular count e.g. a number. Before refactoring there was also `Count` word around. But here we use `counts` which implicitly assumes many numbers e.g. list of numbers which is actually not. Could be confusing.